hal-x86_64/src/framebuffer.rs

```diff
@@ -76,6 +76,18 @@ where
     fn set_pixel(&mut self, x: usize, y: usize, color: RgbColor) -> &mut Self {
         self.set_pixel_rgb(x, y, color)
     }
+
+    fn scroll_vert(&mut self, amount: isize) -> &mut Self {
+        if amount < 0 {
+            todo!("eliza: handle negatives!")
+        }
+        let amount_px = (amount as usize * self.cfg.line_len) * self.cfg.px_bytes;
+        let buf = self.buf.as_mut();
+        let len = buf.len();
+        buf.rotate_left(amount_px);
+        buf[(len - amount_px)..].fill(0);
```

Comment:
This feels like it'd be a bit inefficient to implement, as we're using the more expensive `rotate_left` operation which tries to preserve the rotated out pixels, then immediately clobbering them. Perhaps we should use instead use `copy_within`?

```rust
buf.copy_within(amount_pix.., 0);
buf[(len - amount_pix)..].fill(0);
```

Reply:
TIL about `copy_within`, that's much nicer!